src/Core/Entities/User.cs

```diff
@@ -211,6 +212,25 @@ public int GetSecurityVersion()
         return SecurityVersion ?? 1;
     }
 
+    public bool IsSetupForV2Encryption()
```

Comment:
🎨 (nit): I find this mostly clear, but there is a small amount of ambiguity in this name. `IsSetupFor` can imply they are setup for a potential migration to V2 encryption, but we actually mean that they are already on V2 encryption if this method returns true. Something like `HasV2Encryption` might clarify this but also a `<summary>` on what the method is intending could work as well.